int32     message_code
int32     id
bool[]   available_targets
